### ISSUE TITLE
New win_capability module

### ISSUE THREAD

**Issue by RRohi** [ansible-collections/ansible.windows#96]
##### SUMMARY
Would be nice if there was a module to configure (enable/disable) Windows Capabilities, similar to win_optional_feature module.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

I guess the new module's name would be `win_capability`?

##### ADDITIONAL INFORMATION

The module would replicate the functionality of the cmdlets found via this query:
```powershell
Get-Command -Module Dism -Noun WindowsCapability
```

It would allow people to enable, or disable Windows Capabilities, along with Optional Features and Features (on Servers).

Since Capabilities have version numbers in their names, we should be able to use wildcards, or provide names partially.

Below is an example how I would add RSAT Windows Capabilities with PowerShell and a mockup of an Ansible task where I would disable a couple of Capabilities that are enabled by default.

The asterisk after "{{ item }}" is clearly not correctly placed, I don't know how it should work, but it should indicate that it would be best if the name parameter would accept partial Capability names.

```powershell
Get-WindowsCapability -Online -Name Rsat.* | Add-WindowsCapability -Online
```
```yaml
- name: Disable unnecessary Windows Capabilities
  win_capability:
    name: "{{ item }}"*
    state: absent
  loop:
    - 'XPS.Viewer'
    - 'App.Support.QuickAssist'
    - 'Browser.InternetExplorer'
```

**Comment by jborean93** [comment]
This would be nice to have but I would consider implementing it in [community.windows](https://github.com/ansible-collections/community.windows) as it's not something I see us supporting in this collection anytime soon.

**Comment by davetapley** [comment]
@jborean93 I was about to re-create this issue over at [community.windows](https://github.com/ansible-collections/community.windows), but perhaps you can just move it? 🙏🏻

**Comment by aristotelos** [comment]
> This would be nice to have but I would consider implementing it in [community.windows](https://github.com/ansible-collections/community.windows) as it's not something I see us supporting in this collection anytime soon.

This was attempted (https://github.com/ansible-collections/community.windows/issues/270) but rejected by you yourself @jborean93 ... What is the way forward to support installing Windows capabilities in Ansible?

**Comment by robsterlobster1** [comment]
Im also looking for something like this

**Comment by dom6770** [comment]
Any progress?

I quite don't understand why this doesn't exist. We got a module for [win_feature](https://docs.ansible.com/projects/ansible/latest/collections/ansible/windows/win_feature_module.html#ansible-collections-ansible-windows-win-feature-module) and for [win_optional_feature](https://docs.ansible.com/projects/ansible/latest/collections/ansible/windows/win_optional_feature_module.html#ansible-collections-ansible-windows-win-optional-feature-module), but not for win_capability?

**Comment by stevefulme1** [comment]
@jborean93 This request has been open for 5 years and has bounced between `ansible.windows` and `community.windows` without a path forward. The `community.windows` issue (#270) was closed because that collection only accepts bugfixes, which effectively leaves this feature with no home.

`ansible.windows` already ships `win_feature` (Server Features) and `win_optional_feature` (Optional Features) — `win_capability` is the third and final leg of the Windows feature management surface, wrapping the same DISM backend (`Get-WindowsCapability` / `Add-WindowsCapability` / `Remove-WindowsCapability`).

The demand signal is clear: 22+ thumbs-up on the community issue, consistent engagement over 4 years, and a high-value use case in OpenSSH Server provisioning that users currently have to work around with raw `win_shell` calls (losing idempotency, check mode, and diff support).

Would you be open to accepting a PR for a `win_capability` module in this collection? Happy to help get one started or coordinate a contributor if that would be useful.

**Comment by stevefulme1** [comment]
Following up on my own comment — I missed that this was already implemented! @jborean93 merged #885 adding both `win_capability` and `win_capability_info` modules. These will ship in ansible.windows 3.6.0.

This issue can be closed as resolved. Thanks Jordan for getting this done!
